transaction/builder/src/transaction_builder.rs

```diff
@@ -3216,20 +3058,14 @@ pub mod transaction_builder_tests {
             let mut memo_builder = BurnRedemptionMemoBuilder::new([3u8; 64]);
             memo_builder.enable_destination_memo();
 
-            let mut transaction_builder = TransactionBuilder::new(
-                block_version,
-                Amount::new(10, Mob::ID),
-                fog_resolver.clone(),
-            )
-            .unwrap();
-            transaction_builder.add_input(input_credentials.clone());
+            let mut transaction_builder = transaction_builder.clone();
 
             transaction_builder
-                .add_output(Amount::new(100, token_id), &burn_address(), &mut rng)
+                .add_output(Amount::new(490, Mob::ID), &burn_address(), &mut rng)
```

Comment:
The original test used Mob as the fee token and tried to create change in a different token.
Here it got changed to use `token_id` as the fee token and `Mob` as the change.
The amount also got updated so that the `sum of inputs = sum of outputs - change`, even if inputs and outputs are in different token ids.